ext/serializer.c

```diff
@@ -1307,6 +1321,18 @@ static void _serialize_meta(zval *el, ddtrace_span_data *span) {
         EG(exception) = current_exception;
     }
 
+    zval *git_metadata = &span->root->property_git_metadata;
+    if (git_metadata && Z_TYPE_P(git_metadata) == IS_OBJECT) {
+        ddtrace_git_metadata *metadata = (ddtrace_git_metadata *)Z_OBJ_P(git_metadata);
+        if (is_root_span) {
+            add_assoc_str(meta, "_dd.git.commit.sha", zend_string_copy(Z_STR_P(&metadata->property_commit)));
```

Comment:
Please use `ddtrace_convert_to_str` for safety when reading user-mutable data. It will just do a copy when it's a string, but handle the case where somebody assigned something else instead of crashing.
```suggestion
            add_assoc_str(meta, "_dd.git.commit.sha", ddtrace_convert_to_str(&metadata->property_commit));
```